wiki/planning/astar_planning_implementation_guide.md

```diff
@@ -0,0 +1,109 @@
+---
+# Jekyll 'Front Matter' goes here. Most are set by default, and should NOT be
+# overwritten except in special circumstances.
+# You should set the date the article was last updated like this:
+date: 2020-12-06 # YYYY-MM-DD
+# This will be displayed at the bottom of the article
+# You should set the article's title:
+title: A* Implementation Guide
+# The 'title' is automatically displayed at the top of the page
+# and used in other parts of the site.
+---
+## Introduction
+This wiki aims to serve as a brief introduction and implementation guideline for A* search, specifically applied to robot motion planning. This will discuss how the algorithm functions, important design aspects in the pipeline, common knobs to tune, and efficiency considerations. The specific guidance and notes will be mainly related to implementation for an Ackermann steered vehicle, but will be useful for abstracting to other systems as well.
+
+A* search is just one of many algorithms that can be used for search, but note that grid-based search methods such as this are often limited in their effectiveness when used for higher DOF systems. For those problems, it would be worthwhile to consider sampling based planning methods instead.
+
+These notes and guidance are augmented with heavy borrowing from and attribution to Maxim Likhachev’s 16-782 Planning and Decision Making in Robotics course, Fall 2020.
+
+## Overview of A* Algorithm
+A* is a popular search algorithm that is guaranteed to return an optimal path, and during its search for an optimal path, will provably expand the minimum number of states to guarantee optimality. Why is this useful? It allows for far less exploration, and in turn, less computation than a Dijkstra search, which is a greedy search algorithm.
+
+### Important Terms
+- g(s) value - cost of the shortest path from the start state (sstart) to the current state (s) so far
+- h(s) value - estimate of the cost-to-go from the current state (s) to the goal state (sgoal)
+- f(s) value - total estimated cost from the start state (sstart) to the goal state (sgoal)
+- Admissibility - h(s) is an underestimate of the true cost to goal
+- Monotonicity/Consistent - h(s) <= c(s,s’) + h(s’) for all successors s’ of s
+- Optimality - no path exists from the start to the goal with a lower cost within the constraints of the problem
+
+A* works by computing optimal g-values for all states along the search at any point in time.
+
+![](/assets/astar_h_viz.png)
```

Comment:
All asset paths should be relative -> `assets/astar_h_viz.png`.

Drop the first `/` and you'll be good. Please make sure you get them all.